File: AndroidPatternsApp/jmvp-app/src/main/java/com/woowrale/mvp/ui/base/BaseActivity.java
package com.woowrale.mvp.ui.base;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import com.woowrale.mvp.MainApplication;
import com.woowrale.mvp.di.components.MainComponent;

public abstract class BaseActivity extends AppCompatActivity implements BasePresenter.View {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getDaggerMainComponent().inject(this);
    }

    protected MainComponent getDaggerMainComponent() {
        return ((MainApplication) getApplication()).getMainComponent();
    }

    protected abstract void initDagger();

}

File: RxKotlinRetrofit/app/src/main/java/com/woowrale/rxkotlinretrofit/ui/base/BaseActivity.kt
package com.woowrale.rxkotlinretrofit.ui.base

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

abstract class BaseActivity : AppCompatActivity(), BasePresenter.View {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }
}